File: app/src/main/java/com/luisenricke/simpleroomapp/data_java/entity/PetMedicine.java
package com.luisenricke.simpleroomapp.data_java.entity;

import androidx.annotation.Nullable;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

import java.util.Date;

import static com.luisenricke.simpleroomapp.data_java.entity.PetMedicine.SCHEMA.*;

@Entity(tableName = TABLE,
        foreignKeys = {
                @ForeignKey(entity = Pet.class,
                        parentColumns = Pet.SCHEMA.ID,
                        childColumns = PET,
                        onUpdate = ForeignKey.CASCADE,
                        onDelete = ForeignKey.CASCADE),
                @ForeignKey(entity = Medicine.class,
                        parentColumns = Medicine.SCHEMA.ID,
                        childColumns = MEDICINE,
                        onUpdate = ForeignKey.CASCADE,
                        onDelete = ForeignKey.CASCADE)}
)
public class PetMedicine {
    public interface SCHEMA {
        String TABLE = "PetMedicine";
        String ID = "id";
        String PET = "pet_id";
        String MEDICINE = "medicine_id";
        String CREATED_AT = "created_at";
        String UPDATED_AT = "updated_at";
    }

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = ID)
    private int id;
    @ColumnInfo(name = PET)
    private int petId;
    @ColumnInfo(name = MEDICINE)
    private int medicineId;
    @ColumnInfo(name = CREATED_AT)
    private Date created;
    @ColumnInfo(name = UPDATED_AT)
    private Date updated;

    @Ignore
    public PetMedicine() {
    }

    @Ignore
    public PetMedicine(int id, int petId, int medicineId) {
        this.id = id;
        this.petId = petId;
        this.medicineId = medicineId;
        long time = System.currentTimeMillis();
        created = new Date(time);
        updated = new Date(time);
    }

    public PetMedicine(int petId, int medicineId) {
        this.petId = petId;
        this.medicineId = medicineId;
        long time = System.currentTimeMillis();
        created = new Date(time);
        updated = new Date(time);
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getPetId() {
        return petId;
    }

    public void setPetId(int petId) {
        this.petId = petId;
    }

    public int getMedicineId() {
        return medicineId;
    }

    public void setMedicineId(int medicineId) {
        this.medicineId = medicineId;
    }

    public Date getCreated() {
        return created;
    }

    public void setCreated(Date created) {
        this.created = created;
    }

    public Date getUpdated() {
        return updated;
    }

    public void setUpdated(Date updated) {
        this.updated = updated;
    }

    @Override
    public boolean equals(@Nullable Object obj) {
        if (obj == null || obj.getClass() != PetMedicine.class) return false;
        PetMedicine cast = (PetMedicine) obj;
        return cast.id == getId() && cast.medicineId == getMedicineId() && cast.petId == getPetId();
    }
}

File: app/src/main/java/com/luisenricke/simpleroomapp/data/entity/PetMedicine.kt
package com.luisenricke.simpleroomapp.data.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey
import java.util.*

@Suppress("SpellCheckingInspection")
@Entity(
    tableName = PetMedicine.SCHEMA.TABLE,
    foreignKeys = [
        ForeignKey(
            entity = Pet::class,
            parentColumns = arrayOf(Pet.SCHEMA.ID),
            childColumns = arrayOf(PetMedicine.SCHEMA.PET_ID),
            onUpdate = ForeignKey.CASCADE,
            onDelete = ForeignKey.CASCADE
        ),
        ForeignKey(
            entity = Medicine::class,
            parentColumns = arrayOf(Medicine.SCHEMA.ID),
            childColumns = arrayOf(PetMedicine.SCHEMA.MEDICINE_ID),
            onUpdate = ForeignKey.CASCADE,
            onDelete = ForeignKey.CASCADE
        )
    ]
)
data class PetMedicine(
    @ColumnInfo(name = SCHEMA.PET_ID)
    val petId: Int,
    @ColumnInfo(name = SCHEMA.MEDICINE_ID)
    val medicineId: Int,
    @ColumnInfo(name = SCHEMA.CREATED_AT)
    val createdAt: Date,
    @ColumnInfo(name = SCHEMA.UPDATED_AT)
    val updatedAt: Date,
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = SCHEMA.ID)
    val id: Long = 0
) {
    object SCHEMA {
        const val TABLE = "PetMedecine"
        const val ID = "id"
        const val PET_ID = "pet_id"
        const val MEDICINE_ID = "medicine_id"
        const val CREATED_AT = "created_at"
        const val UPDATED_AT = "updated_at"
    }
}